### PR TITLE
Fix typos and add links

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Refs: #240

**Describe the changes made in this pull request**

- Fix typos
- Fix some minor wording
- Add links where I deemed it useful

Note: all changes for one thing should be encapsulated in one commit, so as to make thing easily revertable.

**Instructions to review the pull request**

- Look at `schema-guide.md`
- Decide if all added links are useful and necessary

### CODE REVIEW

**Comment by sdruskat** [comment]
Ready for review (1 dead link in check isn't really dead :roll_eyes:).

File: schema-guide.md [hunk 1]
```diff
@@ -323,7 +323,7 @@ that case, `doi` can be used as shorthand for something like:<br><br>
 
 - **type**: (Array of) [`definitions.license-enum`](#definitionslicense-enum).
 - **required**: `false`
-- **description**: The SPDX license identifier(s) for the license(s) under which the work is made available. When there are multiple
+- **description**: The [SPDX license identifier(s)](https://spdx.dev/ids/) for the license(s) under which the work is made available. When there are multiple
```
Comment (jspaaks) [inline schema-guide.md:326]: Note that over time, the list in this link will become increasingly out of sync with our list though. In the schema, we have a note which permalinks to github, maybe that's clear enough

File: schema-guide.md [hunk 2]
```diff
@@ -1205,9 +1205,9 @@ An entity can represent different types of entities, such as a team, an institut
     ```
     ```yaml
     authors:
-      - family-names: McClane
+      - family-names: Druskat
```
Comment (jspaaks) [inline schema-guide.md:1208]: I guess no 

Welcome to the party pal!!

for you then :)

hashtag nakatomi towers

File: schema-guide.md [hunk 3]
```diff
@@ -3160,13 +3160,13 @@ licenses, it is assumed their relationship is OR, not AND.
         type: generic
     ```
 
-Note that this key should contain notes that may be picked up by some downstream tooling (e.g., reference managers), but not other (e.g., software index).
+Note that the notes provided under this key may be picked up by some downstream tooling (e.g., reference managers), but not others (e.g., a software index).
```
Comment (jspaaks) [inline schema-guide.md:3163]: maybe move this to `description`

**Review by jspaaks — APPROVED** [review]
Good job, left a couple minor comments

File: schema-guide.md (hunk 3 above)
Comment (sdruskat) [reply schema-guide.md:3163]: Fixed in https://github.com/citation-file-format/citation-file-format/pull/285/commits/11ea8a32892ed4bb4b7ebea33f39aa430ee8c750.

File: schema-guide.md (hunk 1 above)
Comment (sdruskat) [reply schema-guide.md:326]: Decided that this is okay for now during a review call.

**Pull request merged**